src/main/java/com/softserve/demo/util/ContractMaker.java

```diff
@@ -26,35 +29,40 @@
     private static final float ABSOLUTE_Y_POSITION = 30f;
     private static final float CUSTOMER_SIGNATURE_X_POSITION = 440f;
     private static final float PROVIDER_SIGNATURE_X_POSITION = 150f;
+    private static final String PATH = System.getProperty("user.dir");
+    private static final String SEPARATOR = System.getProperty("file.separator");
+    private Path fileStorageLocation;
 
     @Value("${contract.template}")
-    private String pathToContractTemplate;
+    private String contractTemplateFile;
     @Value("${contract.folder}")
     private String contractsFolder;
-    @Value("${contract.path}")
-    private String pathToContracts;
 
-    public void createContract(Order order) throws IOException, DocumentException {
-        if (!createFolderIfNoExist(contractsFolder)) {
-            throw new NotFoundException("Contract folder not found");
+    public void createContract(Order order) {
+        createContractFolder(contractsFolder);
+        String contractFileName = getContractFileName(order);
+        log.info("Create contract with name: [{}]", contractFileName);
+
+        try (
+                FileInputStream fileInputStream = new FileInputStream(getContractFile(contractTemplateFile));
+                FileOutputStream fileOutputStream = new FileOutputStream(getContractFile(contractFileName))
+        ) {
+            PdfReader pdfReader = new PdfReader(fileInputStream);
+            PdfStamper pdfDocument = new PdfStamper(pdfReader, fileOutputStream);
+            setContractFields(order, pdfDocument.getAcroFields());
+            PdfContentByte content = pdfDocument.getOverContent(pdfReader.getNumberOfPages());
+            signContract(content, order);
+            pdfDocument.setFormFlattening(true);
+
+            pdfDocument.close();
```

Comment:
This close methods will never be called in case od error in this method. If you are not using try with resources for them, please, use finally block